integration/tests/cook/test_basic.py

```diff
@@ -2564,6 +2564,60 @@ def test_pools_in_default_limit_response(self):
                 resp = util.get_limit(self.cook_url, limit, user)
                 self.assertFalse('pools' in resp.json())
 
+    @pytest.mark.serial
+    def test_submit_hook(self):
+        job_executor_type = util.get_job_executor_type(self.cook_url)
+
+        # Should succeed, demo-plugin accepts jobs by default.
+        job_uuid, resp = util.submit_job(self.cook_url, executor=job_executor_type)
+        self.assertEqual(resp.status_code, 201, msg=resp.content)
+        self.assertEqual(resp.content, str.encode(f"submitted jobs {job_uuid}"))
+        job = util.wait_for_job(self.cook_url, job_uuid, 'completed')
+        self.assertIn('success', [i['status'] for i in job['instances']], json.dumps(job, indent=2))
+        self.assertEqual(False, job['disable_mea_culpa_retries'])
+        self.assertTrue(len(util.wait_for_output_url(self.cook_url, job_uuid)['output_url']) > 0)
+
+        # Tell demo plugin server to reject jobs.
+        fail = {'status': 'rejected', 'message': 'A message'}
+        demo_hook_plugin = os.getenv('DEMO_HOOKS_SERVICE')
+        util.session.post(f'{demo_hook_plugin}/set-submit-status', json=fail)
+
+        # This should now fail to submit.
+        job_uuid, resp = util.submit_job(self.cook_url, executor=job_executor_type)
+        self.assertEqual(resp.status_code, 400, msg=resp.content)
+
+        success = {'status': 'accepted', 'message': 'A message'}
+        util.session.post(f'{demo_hook_plugin}/set-submit-status', json=success)
```

Comment:
Stick this in a finally so we don't accidentally leave the service in a bad state.

Reply:
Good suggestion.